pubsub/pulsar/pulsar.go

```diff
@@ -123,16 +123,30 @@ func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
 		producer = cache.(pulsar.Producer)
 	}
 
-	_, err = producer.Send(context.Background(), &pulsar.ProducerMessage{
-		Payload: req.Data,
-	})
-	if err != nil {
+	msg := parsePublishMetadata(req)
+	if _, err = producer.Send(context.Background(), msg); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// parsePublishMetadata parse publish metadata
+func parsePublishMetadata(req *pubsub.PublishRequest) (msg *pulsar.ProducerMessage) {
+	msg = &pulsar.ProducerMessage{
+		Payload: req.Data,
+	}
+	if val, ok := req.Metadata["deliver_at"]; ok {
+		msg.DeliverAt, _ = time.Parse("2006-01-02 15:04:05", val)
+	}
+	if val, ok := req.Metadata["deliver_after"]; ok {
+		tmp, _ := strconv.Atoi(val)
+		msg.DeliverAfter = time.Duration(tmp)
```

Comment:
Why are we not using `time.ParseDuration(val)` here instead? It seems very unfriendly to the dev to require that they calculate the integrate an integer duration instead of Go-standard duration strings like "3h" used elsewhere in Dapr.